planner/core/rule_column_pruning.go

```diff
@@ -106,7 +109,7 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 		aggrFunc.OrderByItems, cols = pruneByItems(aggrFunc.OrderByItems)
 		selfUsedCols = append(selfUsedCols, cols...)
 	}
-	if len(la.AggFuncs) == 0 {
+	if len(la.AggFuncs) == 0 || (!allFirstRow && allRemainFirstRow) {
```

Comment:
Add comments why we need to add this check

Reply:
Comments are below this line. 
https://github.com/pingcap/tidb/pull/25289/files#diff-ae586e551e5a201a40c83192b7dc23cc2778739ef522fcb02e4fb7fac2125c99R113